File: api/audio-book/src/main/java/org/backend/book/dto/response/PurchasedBookResponse.java
package org.backend.book.dto.response;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.backend.file.dto.FileDto;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
public class PurchasedBookResponse {
    private Long id;
    private Long bookId;
    private String bookName;
    private String bookAuthor;
    private FileDto coverFile;
    private LocalDateTime purchasedAt;
    private Boolean isActive;
    private Boolean expired;
}

File: mobile_client/lib/src/components/library/models/purchased_book_response.dart
class PurchasedBookResponse {
  final int id;
  final int bookId;
  final String bookName;
  final String bookAuthor;
  final String? coverUrl;
  final DateTime? purchasedAt;
  final bool? isActive;
  final bool? expired;

  PurchasedBookResponse({
    required this.id,
    required this.bookId,
    required this.bookName,
    required this.bookAuthor,
    this.coverUrl,
    this.purchasedAt,
    this.isActive,
    this.expired,
  });

  factory PurchasedBookResponse.fromJson(Map<String, dynamic> json) {
    String? finalCoverUrl;
    
    if (json['coverFile'] != null) {
      final coverFile = json['coverFile'];
      if (coverFile is Map<String, dynamic>) {
        finalCoverUrl = coverFile['fileUrl'] ?? coverFile['filePath'];
      }
    }

    DateTime? parsedDate;
    if (json['purchasedAt'] != null) {
      parsedDate = DateTime.tryParse(json['purchasedAt'].toString());
    }

    return PurchasedBookResponse(
      id: json['id'] ?? 0,
      bookId: json['bookId'] ?? 0,
      bookName: json['bookName'] ?? '',
      bookAuthor: json['bookAuthor'] ?? '',
      coverUrl: finalCoverUrl,
      purchasedAt: parsedDate,
      isActive: json['isActive'],
      expired: json['expired'],
    );
  }
}
